Source/WebKit/UIProcess/API/glib/WebKitApplicationInfo.h.in

```diff
@@ -17,15 +17,26 @@
  * Boston, MA 02110-1301, USA.
  */
 
+#if PLATFORM(GTK)
 #if !defined(__WEBKIT2_H_INSIDE__) && !defined(WEBKIT2_COMPILATION)
 #error "Only <webkit2/webkit2.h> can be included directly."
 #endif
+#elif PLATFORM(WPE)
+#if !defined(__WEBKIT_H_INSIDE__) && !defined(WEBKIT2_COMPILATION)
+#error "Only <wpe/webkit.h> can be included directly."
+#endif
+#endif
```

Comment:
Now that we are going to generate the headers, I wonder if we could simplify them even more. For example this could be something like `@PLATFORM_SINGLE_HEADER_CHECK@`, or something like that, that is substituted by the platform check. We would probably need a script that process the header and calls unifdef as the last step.

Reply:
Good idea, especially because this is going to become even more complicated if I succeed in dropping the `2` from the GTK API version, in which case we'll have three conditions here.

We *might* be able to avoid a new script if we define `PLATFORM_SINGLE_HEADER_CHECK` in a CMake variable and use configure_file().

Can we do this in the future rather than now, though? If it turns out to be very simple then I might submit a follow-up imminently, but if not, it's not important enough to block on IMO.

Also, keep in mind the required checks are different for UI process vs. InjectedBundle, and files that are shared will need yet another variant to allow both.

Reply:
Sure, we can improve it in follow up patches.

Reply:
Agreed to do this in a follow-up patch.

Reply:
Wound up doing it in this PR anyway.